lusherr/frame.go

```diff
@@ -0,0 +1,73 @@
+package lusherr
+
+import (
+	"errors"
+	"fmt"
+	"runtime"
+)
+
+// originError is used to give any error an origin frame.
+type originError struct {
+	err   error
+	frame runtime.Frame
+}
+
+func (e originError) Error() string {
+	return fmt.Sprintf("%v", e.err)
+}
+
+// Locate the frame of the error.
+func (e originError) Locate() runtime.Frame {
+	return e.frame
+}
+
+// Originate the error from a frame.
+func (e originError) Originate(frame runtime.Frame) error {
+	e.frame = frame
+	return e
+}
+
+// Locator defines behavior for locating an error frame.
+type Locator interface {
+	Locate() runtime.Frame
+}
+
+// Locate where an error was raised.
+func Locate(err error) (runtime.Frame, bool) {
+	var frame runtime.Frame
+	switch err := err.(type) {
+	case Locator:
+		return err.Locate(), true
+	default:
+		if err := errors.Unwrap(err); err != nil {
+			return Locate(err)
+		}
+	}
+	return frame, false
+}
+
+// Originator defines behavior for defining an origin frame for an error.
+type Originator interface {
+	Originate(runtime.Frame) error
```

Comment:
Just kinda unhappy with this name here.
Should it simply be `Wrap` ?

Even the description in the readme employs that term instead

Reply:
I don’t want to confuse it with Unwrap or get close to that territory. But I get your point. Got any ideas?

Reply:
That's a fair point.
Since it makes the error traceable by wrapping in the caller's Frame, I would suggest we call it `Trace`?

Reply:
Interesting. I’ll think about it and get back to you. Thank you!